docs/troubleshooting/Performance.mdx

```diff
@@ -79,6 +79,86 @@ module.exports = {
 
 See [Glob pattern in parser's option "project" slows down linting](https://github.com/typescript-eslint/typescript-eslint/issues/2611) for more details.
 
+## Changes to `extraFileExtensions` with `projectService` in your ESLint options
+
+Using different [`extraFileExtensions`](../packages/Parser.mdx#extrafileextensions) for different files with [`projectService`](../packages/Parser.mdx#projectservice) will cause a full project reload.
```

Comment:
[Docs]

> will cause a full project reload

This is a little vague out of context. Can you be a bit more specific? In what scenario(s) does it cause the reload? How does that impact code? How would a user observe this?

Reply:
Changing `extraFileExtensions` with tsservers `setHostConfiguration` causes tsserver to initiate a project reload to update the project structure with the extension changes: https://github.com/microsoft/TypeScript/blob/b1c52c53cc6c8cf35b19accfc3e29916489c821e/src/server/editorServices.ts#L3781

Tsserver does not appear to do any checks if the old file extensions match the new file extensions.

The current implementation of `typescript-estree` updates the file extension for every file when `extraFileExtensions` is set.  This compounds with plugins like `vue-eslint-parser` where each template fragment triggers this.

This PR caches the last set `extraFileExtensions` to minimize these project reloads.

Users can further optimize, which is why I added it to the docs, by making sure `extraFileExtensions` is only set once.

I can merge in main now that the logging PR was accepted to expose tsserver logs and provide some examples.

A user could look for this log message as an example: https://github.com/microsoft/TypeScript/blob/b1c52c53cc6c8cf35b19accfc3e29916489c821e/src/server/editorServices.ts#L3825

Reply:
Oh, just clarifying, I meant those questions as rhetorical. They're things that the docs should answer. I'm requesting additions to the docs. Sorry for not being clear 😅

Reply:
Additions have been added.  Let me know if additional detail is needed in any area.

I added my [own comment](https://github.com/typescript-eslint/typescript-eslint/pull/9336#discussion_r1643353701) to point out a creative liberty taken to help with readability.

I can revert for accuracy if preferred.

Reply:
These additions are fantastic! I hadn't thought of adding an FAQ article, but it makes a ton of sense. Strong +1 😄